protobuf/src/test/java/io/grpc/protobuf/ProtoUtilsTest.java

```diff
@@ -60,6 +62,17 @@ public void testJsonRoundtrip() throws Exception {
     assertEquals(proto, marshaller.parse(is));
   }
 
+  @Test
+  public void testJsonMarshallerFailToPrint() {
+    Marshaller<Any> marshaller = ProtoUtils.jsonMarshaller(Any.getDefaultInstance());
+    try {
+      marshaller.stream(Any.newBuilder().setValue(ByteString.copyFromUtf8("invalid")).build());
```

Comment:
I had to actually run the test to see why this fails - perhaps change the string to something like "invalid (no type url)" to make this a little more obvious?

Reply:
Here just sets the invalid bytes to [value](https://github.com/google/protobuf/blob/c93174346149f259ebc85fdb46826fcf84abb07d/src/google/protobuf/any.proto#L153).

I have no other ideas for here...

Reply:
I meant just providing a hint as to *why* the proto is invalid. In general, I believe it can be fine to set the value of Any to a string (if the typeUrl is set to `type.googleapis.com/google.protobuf.StringValue`), so knowing why this should throw an exception requires knowing details about the Any type. Looking at this a little more, it looks like using Any-type with JsonFormat requires providing a TypeRegistry to the JsonFormat Parser even for well-known types, but that's not obvious in this test either.

I just intended the following minor change to what you had originally, so it's more clear why the message should fail to parse:

```
marshaller.stream(Any.newBuilder().setValue(ByteString.copyFromUtf8("invalid (no type url)")).build());
```

Reply:
I really appreciate your explanation. Now I understand what should I do.